### ISSUE TITLE
How to handle these new jGroups clusters that end in 8888

### ISSUE THREAD

**Issue by cstockloew** [universAAL/middleware#449]
_Originally Opened: @Alfiva (2014-11-19 13:31:49_)
_Originally Closed: 2014-11-19 17:57:30_

I have tried the direct connection of Android instance and PC instance in version 3.2.0 and I am having problems when it used to work just ok in previous versions.

In particular I would like to know if this message in the log (and I get tons of these) of the PC has anything to do with it:

a71e148e-17d8-4bb0-bac8-614017345b34: discarding discovery request for cluster 'mw.bus.context.osgi' from b5ada36a-420d-399e-b847-94feede1de7e; our cluster name is 'mw.modules.aalspace.osgi8888'. Please separate your clusters cleanly.

Since I don't remember seeing the AALSpace ID appended to the module ID in the cluster: mw.modules.aalspace.osgi8888. Notice that the ones generated in Android are (and as far as I remember have always been) something like: mw.modules.aalspace.osgi Without the 8888 (AALSpaceID) in the end.

Has there been any change regarding this that I should take into account? Can this be the reason that I cannot make Android connect to PC?

--

From: _this issue has been automatically imported from our old issue tracker_

**Comment by cstockloew** [comment]
_Original author: @Alfiva (2014-11-19 13:43:02)_

I just checked: With the exact same configuration, but using version 3.1.0 in the PC, it all works perfect (there is serialization so there is no incompatibility for having 3.2.0 in Android and 3.1.0 in PC).

I also get some of those &quot;Please separate your clustersr cleanly&quot; messages, but they are only for the same long-address in both sides (in other words, it discards its own clusters): Whereas in my first post it is ...5b34 and ...de7e, in my 3.1.0 attempt it is ...3eb2 and ...3eb2.

And of course cluster IDs dont have 8888 in the end.

**Comment by cstockloew** [comment]
_Original author: @stefanolenzi-cnr (2014-11-19 14:18:30)_

Adding the SpaceId at the end of the bus name was introduced with 3.2.0 for enabling the creation of several AALSpace on the same network without having to change the port and/or ip used for multicasting messages

if I recall correctly the old implementation was performing a startWith() matching, thus the 3.1.0 should receive the message anyhow but you should backport the change from PC version to Android.

**Comment by cstockloew** [comment]
_Original author: @Alfiva (2014-11-19 16:07:17)_

Any hint on where exactly is this addition performed? I have a few MW modules branched and some other replaced by Android MW, so I guess the code must in one of those. Where exactly?

**Comment by cstockloew** [comment]
_Original author: @Alfiva (2014-11-19 17:57:30)_

Thanks for the hint through Skype (AALSpaceManagerImpl revision 3996, as explained in issue #410).

It is now fixed. I have retro-fixed the 3.2.0 tag of the Android MW, fixed the current trunks, and committed the updated apk.

The reason for this issue was that I had overlooked a single line during the merge of the android branch of the AALSpaceManager module.
